src/ioctls/vm.rs

```diff
@@ -1358,6 +1426,68 @@ mod tests {
         assert!(vm_fd.unregister_irqfd(&evtfd3, 5).is_ok());
     }
 
+    #[test]
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    fn test_set_irq_line() {
+        let kvm = Kvm::new().unwrap();
+        let vm_fd = kvm.create_vm().unwrap();
+
+        assert!(vm_fd.create_irq_chip().is_ok());
+
+        assert!(vm_fd.set_irq_line(4, true).is_ok());
+        assert!(vm_fd.set_irq_line(4, false).is_ok());
+        assert!(vm_fd.set_irq_line(4, true).is_ok());
+    }
+
+    #[test]
+    #[cfg(target_arch = "aarch64")]
+    fn test_set_irq_line() {
+        let kvm = Kvm::new().unwrap();
+        let vm_fd = kvm.create_vm().unwrap();
+        // Create a vcpu for test case 2 of the KVM_IRQ_LINE API on aarch64.
+        vm_fd.create_vcpu(0).unwrap();
+
+        // Create the vGIC device.
+        let vgic_fd = create_gic_device(&vm_fd, 0);
+
+        // Dummy interrupt for testing on aarch64.
+        let nr_irqs: u32 = 128;
+
+        // We need to tell the kernel how many irqs to support with this vgic.
+        let vgic_attr = kvm_bindings::kvm_device_attr {
+            group: kvm_bindings::KVM_DEV_ARM_VGIC_GRP_NR_IRQS,
+            attr: 0,
+            addr: &nr_irqs as *const u32 as u64,
+            flags: 0,
+        };
+        assert!(vgic_fd.set_device_attr(&vgic_attr).is_ok());
+        // Finalize the GIC.
+        let vgic_attr = kvm_bindings::kvm_device_attr {
+            group: kvm_bindings::KVM_DEV_ARM_VGIC_GRP_CTRL,
+            attr: u64::from(kvm_bindings::KVM_DEV_ARM_VGIC_CTRL_INIT),
+            addr: 0,
+            flags: 0,
+        };
+        assert!(vgic_fd.set_device_attr(&vgic_attr).is_ok());
```

Comment:
This code can also be moved to a function and re-used in `test_register_unregister_irqfd`. We can call it something like `request_gic_init`. This name seems to be closer to the documentation than "finalize_gic":
> KVM_DEV_ARM_VGIC_CTRL_INIT
>       request the initialization of the VGIC, no additional parameter in
>       kvm_device_attr.addr.

Reply:
Done :)